src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -51,18 +53,57 @@ private static VoltageLevelMapData toMapData(VoltageLevel voltageLevel, boolean
         VoltageLevelMapData.VoltageLevelMapDataBuilder builder = VoltageLevelMapData.builder()
                 .id(voltageLevel.getId())
                 .topologyKind(voltageLevel.getTopologyKind());
+
         if (includeDetails) {
             builder.name(voltageLevel.getOptionalName().orElse(null))
                     .substationId(voltageLevel.getSubstation().map(Substation::getId).orElse(null))
-                    .nominalVoltage(voltageLevel.getNominalV());
-
+                    .nominalVoltage(voltageLevel.getNominalV())
+                    .lowVoltageLimit(voltageLevel.getLowVoltageLimit())
+                    .highVoltageLimit(voltageLevel.getHighVoltageLimit());
             if (voltageLevel.getTopologyKind().equals(TopologyKind.NODE_BREAKER)) {
-                builder.busbarSections(voltageLevel.getNodeBreakerView().getBusbarSectionStream().map(NetworkMapService::toMapData).collect(Collectors.toList()));
+                mapVoltageLevelSwitchKindsAndSectionCount(builder, voltageLevel);
+            }
+            IdentifiableShortCircuit identifiableShortCircuit = voltageLevel.getExtension(IdentifiableShortCircuit.class);
+            if (identifiableShortCircuit != null) {
+                builder.ipMin(identifiableShortCircuit.getIpMin());
+                builder.ipMax(identifiableShortCircuit.getIpMax());
             }
         }
         return builder.build();
     }
 
+    private static void mapVoltageLevelSwitchKindsAndSectionCount(
+            VoltageLevelMapData.VoltageLevelMapDataBuilder builder, VoltageLevel voltageLevel) {
+        AtomicInteger busbarCount = new AtomicInteger(1);
+        AtomicInteger sectionCount = new AtomicInteger(1);
+        AtomicBoolean warning = new AtomicBoolean(false);
+        voltageLevel.getNodeBreakerView().getBusbarSections().forEach(bbs -> {
+            var pos = bbs.getExtension(BusbarSectionPosition.class);
+            if (pos != null) {
+                if (pos.getBusbarIndex() > busbarCount.get()) {
+                    busbarCount.set(pos.getBusbarIndex());
+                }
+                if (pos.getSectionIndex() > sectionCount.get()) {
+                    sectionCount.set(pos.getSectionIndex());
+                }
+            } else {
+                warning.set(true);
+            }
+        });
+        builder.isPartiallyCopied(warning.get());
+        if (!warning.get()) {
+            builder.busbarCount(busbarCount.get());
+            builder.sectionCount(sectionCount.get());
+            SwitchKind[] switchKinds = new SwitchKind[sectionCount.get() - 1];
+            Arrays.fill(switchKinds, SwitchKind.DISCONNECTOR);
+            builder.switchKinds(Arrays.asList(switchKinds));
```

Comment:
builder.switchKinds(new ArrayList<SwitchKind>(Collections.nCopies(sectionCount.get() - 1, SwitchKind.DISCONNECTOR));    in one single line ?

Reply:
Done